src/app/routes/share.ts

```diff
@@ -1,36 +1,275 @@
+    }
+
+    if (Validator.isInvalidString(code)) {
+      throw createHttpError(400, 'Code must be a valid string');
+    }
+
+    if (Validator.isInvalidPositiveNumber(directoryId)) {
+      throw createHttpError(400, 'Directory ID is not valid');
+    }
+
+    if (Validator.isInvalidUnsignedNumber(offset)) {
+      throw createHttpError(400, 'Offset is not valid');
+    }
+
+    if (Validator.isInvalidPositiveNumber(limit)) {
+      throw createHttpError(400, 'Limit is not valid');
+    }
+
+    return this.services.Share.getSharedDirectoryFiles(directoryId, Number(offset), Number(limit), token, code)
+      .then((results: unknown) => {
+        res.status(200).json(results);
+      })
+      .catch((err: Error) => {
+        res.status(500).json({
+          error: err.message
+        });
+      });
+  }
+
+  public async getSharedDirectoryFolders(req: Request, res: Response): Promise<void> {
+    const { token, directoryId, offset, limit } = req.query;
+
+    if (Validator.isInvalidString(token)) {
+      throw createHttpError(400, 'Token must be a valid string');
+    }
+
+    if (Validator.isInvalidPositiveNumber(directoryId)) {
+      throw createHttpError(400, 'Directory ID is not valid');
+    }
+
+    if (Validator.isInvalidUnsignedNumber(offset)) {
+      throw createHttpError(400, 'Offset is not valid');
+    }
+
+    if (Validator.isInvalidPositiveNumber(limit)) {
+      throw createHttpError(400, 'Limit is not valid');
+    }
+
+    return this.services.Share.getSharedDirectoryFolders(directoryId, Number(offset), Number(limit), token)
+      .then((results: unknown) => {
+        res.status(200).json(results);
+      })
+      .catch((err: Error) => {
+        res.status(500).json({
+          error: err.message
+        });
+      });
+  }
+
+  private logReferralError(userId: unknown, err: Error) {
+    if (!err.message) {
+      return this.logger.error('[STORAGE]: ERROR message undefined applying referral for user %s', userId);
+    }
+
+    if (err instanceof ReferralsNotAvailableError) {
+      return;
+    }
+
+    return this.logger.error('[STORAGE]: ERROR applying referral for user %s: %s', userId, err.message);
+  };
 }
 
-export default (router: Router, service: any) => {
-  const controller = new ShareController(service);
+export default (router: Router, service: any,) => {
+  const Logger = logger.getInstance();
+  const controller = new ShareController(service, Logger);
   const sharedAdapter = sharedMiddlewareBuilder.build(service);
 
-  router.get('/share/list', passportAuth, sharedAdapter, controller.listShares.bind(controller));
-  router.get('/share/:shareId/folder/:folderId', controller.getSharedFolderSize.bind(controller));
+  router.get('/share/list', passportAuth, sharedAdapter,
+    controller.listShares.bind(controller)
+  );
+  router.get('/share/:shareId/folder/:folderId',
+    controller.getSharedFolderSize.bind(controller)
+  );
+  router.post('/storage/share/file/:id', passportAuth, sharedAdapter,
```

Comment:
For code maintainability, paths related to shares will be better under `/share` path, distributing paths outside its controllers will be painful when the number of endpoints grows. Also, spotting an undesired path match due to a bug introduced by any of us will be much more difficult.

This is a comment for marking this for the next refactor which will be this + SDK required changes

Reply:
If this endpoint is creating a share, an approach with would be near to REST, could be:

POST /share

As we have a differentiation between shared resources, a possible approach for these routes could be found in the next comment